static/app/views/settings/components/settingsBreadcrumb/context.tsx

```diff
@@ -0,0 +1,74 @@
+import {createContext, useCallback, useContext, useEffect, useState} from 'react';
+import {PlainRoute} from 'react-router';
+
+import getRouteStringFromRoutes from 'sentry/utils/getRouteStringFromRoutes';
+import {useRoutes} from 'sentry/utils/useRoutes';
+
+type UpdateData = {
+  routes: PlainRoute[];
+  title: string;
+};
+
+type PathMap = Record<string, string>;
+
+type ContextState = {
+  pathMap: PathMap;
+  updateRouteMap: (update: UpdateData) => void;
+};
+
+const BreadcrumbContext = createContext<ContextState | undefined>(undefined);
+
+type ProviderProps = {
+  children: React.ReactNode;
+};
+
+function BreadcrumbProvider({children}: ProviderProps) {
+  const routes = useRoutes();
+  const [pathMap, setPathMap] = useState<PathMap>({});
+
+  // When our routes change update the path mapping
+  useEffect(
+    () =>
+      setPathMap(oldPathMap => {
+        const routePath = getRouteStringFromRoutes(routes);
+        const newPathMap = {...oldPathMap};
+
+        for (const fullPath in newPathMap) {
+          if (!routePath.startsWith(fullPath)) {
+            delete newPathMap[fullPath];
+          }
+        }
+
+        return newPathMap;
+      }),
+    [routes, setPathMap]
+  );
+
+  /**
+   * Updates the the breadcrumb title at the provided route object
```

Comment:
Nice catch @markstory !

Reply:
tyvm. Still need to fix tests.

After this I'm going to kill off reading the `name` property from the route, and instead have a different route component that renders and passes along the breadcrumb to context with some other hook.

Since in the future, you cannot read props off the router matche objects.